lib/choria/colt/data_structurer.rb

```diff
@@ -0,0 +1,28 @@
+module Choria
+  class Colt
+    module DataStructurer
+      def self.structure(results)
+        results.map do |res|
+          # data.stdout seems to always be JSON, so parse it once.
+          res[:result] = JSON.parse res[:data][:stdout]
+          res[:data].delete :stdout
+
+          # On one side, data.stderr is filled by the remote execution stderr.
+          # On the other side, error description is in JSON (ie. '_error')
+          # So merge data.stderr in '_error'.'details'
+          unless res[:data][:stderr].empty?
+            raise NotImplementedError, 'What to do when res[:data][:stderr] contains something?' if res[:result]['_error'].empty?
```

Comment:
Not raise an error :smile: 

Just ignore it, stderr is indeed for reporting errors but it such a case the command is supposed to have a non-null exit code and it is the only thing the caller should be dealing with.

If the program wants to display progress, prompt the user for some value, etc it will do so on stderr so that the user can see this even if they redirect stdout or pipe it in another command.

If some messages are output on stderr and the command exited successfully, we do not care about it.  If the command fail, reporting stderr to the end-user makes sense.

Reply:
I will keep this exception during development to catch a potential override and remove it if none found, thanks for review.